docs/eventing/sources/pingsource.md

```diff
@@ -1,196 +1,257 @@
 ---
 title: "PingSource"
 linkTitle: "PingSource"
-weight: 31
+weight: 10
 type: "docs"
 ---
 
 ![version](https://img.shields.io/badge/API_Version-v1beta2-red?style=flat-square)
 
-A PingSource produces events with a fixed payload on a specified cron schedule.
+This topic explains event-driven architecture in terms of producers and consumers, and shows how to
+configure PingSource as an event source targeting a Knative service.
 
-## Installation
+## Before you begin
 
-The PingSource source type is enabled by default when you install Knative Eventing.
+1. Set up [Knative Serving](../../../serving).
+1. Set up [Knative Eventing](../../../eventing).
 
-## Example
+## About Producers and Consumers
 
-This example shows how to send an event every minute to a Event Display Service.
+### Producers
 
-### Creating a namespace
+Producers create events. Using GitHub as an example, the producer is the GitHub webhook that sends
+information when a pull request is created.
+Essentially, you ask the producer for events that they are capable of sending and you tell the
+producer where to send those events.
 
-Create a new namespace called `pingsource-example` by entering the following
-command:
+Producers send the event data per specifications that make it easy for consumers to handle it.
+Knative uses the CloudEvents specifications by default.
+For more information, see the [Cloud Events](https://cloudevents.io/) website.
 
-```shell
-kubectl create namespace pingsource-example
-```
+### Consumers
+
+Consumers consume the data that producers create. Consumers are usually functions.
 
-### Creating the Event Display Service
+In the GitHub example above, a consumer might update a ticket in the ticketing system with the data
+from the pull request.
 
-In this step, you create one event consumer, `event-display` to verify that
-`PingSource` is properly working.
 
-To deploy the `event-display` consumer to your cluster, run the following
-command:
+## Create a consumer
+
+To verify that the PingSource is working, create a simple Knative service that receives CloudEvents
+and writes them to standard out.
+
+{{< tabs name="create-service" default="YAML" >}}
+{{% tab name="YAML" %}}
+Create the service from standard input by running:
 
 ```shell
-kubectl -n pingsource-example apply -f - << EOF
-apiVersion: apps/v1
-kind: Deployment
+cat <<EOF | kubectl create -f -
```

Comment:
not sure what the reasoning is for using `cat` here as opposed to
```
kubectl apply -f - <<EOF
```
Slack comment from @evankanderson :
"I think we should do the following:
- Use kubectl apply for files/objects that the user creates — it works for both “create” and “update”, and the source of truth is their local files.
- Use kubectl edit for files which are shipped as part of the Knative software, like the serving/eventing ConfigMaps."

Reply:
Added something to the style guide for this

Reply:
I think there are two points:

1. This is an "unnecessary cat" shell construction; I'd agree with the format that Ashleigh suggests
2. Prefer `kubectl apply` over `kubectl create` unless you are using `metadata.generateName`, which is probably not a good choice in a tutorial, because you can't predict the actual object name.